crates/wasm_interp/src/instance.rs

```diff
@@ -191,17 +209,35 @@ impl<'a, I: ImportDispatcher> Instance<'a, I> {
         let fn_index = {
             let mut export_iter = module.export.exports.iter();
             export_iter
+                // First look up the name in exports
                 .find_map(|ex| {
                     if ex.ty == ExportType::Func && ex.name == fn_name {
                         Some(ex.index)
                     } else {
                         None
                     }
                 })
-                .ok_or(format!(
-                    "I couldn't find an exported function '{}' in this WebAssembly module",
-                    fn_name
-                ))?
+                .or_else(|| {
+                    // Then look it up in the debug info!
+                    // This is non-spec behaviour that Wasm3 seems to implement,
+                    // and that our wasm_linking tests accidentally rely on!
+                    let mut names = module.names.function_names.iter();
+                    names.find_map(
+                        |(index, name)| {
+                            if *name == fn_name {
+                                Some(*index)
+                            } else {
+                                None
+                            }
+                        },
```

Comment:
this can be written as `|(index, name)| (*name == fn_name).then_some(*index)`

Reply:
Wow, I would never have thought of even looking for something like that.